### PR TITLE
Happy thoughts API - Josephine

### PR DESCRIPTION
**Pull request by prosperpine**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,29 +1,90 @@
...
+  const { thoughtId } = req.params;
+  try {
+    const likedThought = await Thought.updateOne(
+      { _id: thoughtId },
+      { $inc: { hearts: 1 } }
+    );
+    res.json(likedThought);
+  } catch (err) {
+    res
+      .status(401)
```
Comment (HIPPIEKICK) [inline server.js:82]: Status 401 means unauthorized. Could 400 (bad request) be better?

File: server.js [hunk 2]
```diff
@@ -1,29 +1,90 @@
...
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Hello happy thoughts');
+});
+
+// Get the thoughts from database
+app.get('/thoughts', async (req, res) => {
```
Comment (HIPPIEKICK) [inline server.js:44]: The documentation says this route should be "/"

File: server.js [hunk 3]
```diff
@@ -1,29 +1,90 @@
...
+  } catch (err) {
+    res.status(400).json({
+      message: 'Could not find any happy thoughts',
+      errors: err.errors,
+    });
+  }
+});
+
+// Add new thoughts to the database
+app.post('/thoughts', async (req, res) => {
```
Comment (HIPPIEKICK) [inline server.js:57]: And this too: "/"

File: server.js [hunk 4]
```diff
@@ -1,29 +1,90 @@
...
+  } catch (err) {
+    res.status(400).json({
+      message: 'Could not save happy thought',
+      errors: err.errors,
+    });
+  }
+});
+
+// Add likes to an existing thought
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
```
Comment (HIPPIEKICK) [inline server.js:72]: The documentation says this route should be `/:thoughtId/like`

**Comment by prosperpine** [comment]
Hi,

I've changed this now and made a new commit. :)

//Josephine

Den mån 25 maj 2020 kl 15:09 skrev Matilda Arvidsson <
notifications@github.com>:

**Comment by HIPPIEKICK** [comment]
So fast! 🤩 Just make sure you don't have two endpoints with the same name :)

**Comment by prosperpine** [comment]
Oh yes of course!:)  The first one was for testing, deleted it now.

Den mån 25 maj 2020 kl 15:26 skrev Matilda Arvidsson <
notifications@github.com>:

**Comment by HIPPIEKICK** [comment]
> Looks good! Just update the routes to match the requested routes in the documentation:
> 
> ```
> GET /
> POST /
> POST /:thoughtId/like
> ```
> 
> Keep up the good work!

Never mind this! `/thoughts` is also fine

**Pull request closed without merging**